The server with MemDB goes down when retrieving an ID that is not in the GetDocument API

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**:

The server with MemDB goes down when retrieving an ID that is not in the GetDocument API.

```
panic: interface conversion: interface {} is nil, not *database.DocInfo

goroutine 58 [running]:
github.com/yorkie-team/yorkie/server/backend/database/memory.(*DB).FindDocInfoByID(0x1024844?, {0x5?, 0xfcf740?}, {0xc0004a4270, 0x18})
	/app/server/backend/database/memory/database.go:401 +0x35f
github.com/yorkie-team/yorkie/server/documents.GetDocumentSummary({0x12ce6e0, 0xc0004aef30}, 0xc00007d6d0, {0xc0004a4270?, 0xc00013fa88?})
	/app/server/documents/documents.go:63 +0x4f
github.com/yorkie-team/yorkie/server/admin.(*Server).GetDocument(0xfcf740?, {0x12ce6e0?, 0xc0004aef30?}, 0xc0004ae630?)
	/app/server/admin/server.go:182 +0x3e
github.com/yorkie-team/yorkie/api._Admin_GetDocument_Handler({0xf9a1c0?, 0xc0000be0d8}, {0x12ce6e0, 0xc0004aef30}, 0xc0004f63c0, 0x0)
	/app/api/admin.pb.go:729 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003d5340, {0x12d2e88, 0xc0001e8ea0}, 0xc0000a86c0, 0xc0000bc690, 0x194ca60, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0003d5340, {0x12d2e88, 0xc0001e8ea0}, 0xc0000a86c0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:1616 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:919 +0x28a
```

**What you expected to happen**:

The server should return the DocumentNotFound Error.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Operating system:
- Browser and version:
- Yorkie version (use `yorkie version`):
- Yorkie JS SDK version:

--- a/server/backend/database/memory/database.go
+++ b/server/backend/database/memory/database.go
@@ -398,6 +398,10 @@ func (d *DB) FindDocInfoByID(
 		return nil, err
 	}
 
+	if raw == nil {
+		return nil, fmt.Errorf("%s: %w", id, database.ErrDocumentNotFound)
+	}
+
 	docInfo := raw.(*database.DocInfo)
 	return docInfo.DeepCopy(), nil
 }

--- a/server/backend/database/memory/database_test.go
+++ b/server/backend/database/memory/database_test.go
@@ -254,4 +254,12 @@ func TestDB(t *testing.T) {
 		assert.NoError(t, err)
 		assertKeys(nil, emptyInfos)
 	})
+
+	t.Run("FindDocInfoByID test", func(t *testing.T) {
+		localDB, err := memory.New()
+		assert.NoError(t, err)
+
+		_, err = localDB.FindDocInfoByID(context.Background(), notExistsID)
+		assert.ErrorIs(t, err, database.ErrDocumentNotFound)
+	})
 }

